Feature/re former

jessicafarias opened the pull request:
The purpose of this project was to learn to create forms using pure HTML then transforming it to embedded ruby forms, in this project, we created an app named re-former in which you can create a new User and edit it.

aracelicaes committed:
Created Re-Former App
Created New Rails App Re-Former
Created it using postgresql database

jessicafarias committed:
Set up the Back-End
Create model and controller for User
Adding only new and create verbs for User
Adding validations to each User params
Adding new.html.erb for users
Create and migrate db

aracelicaes committed:
Created HTML Form
Added HTML form in new.html.erb
Added Create method in user_controller
Add private #user_params in controller
Created db and migrated.

jessicafarias committed:
Adding form_with
Adding form_for(commented)
Edit create and new methods in user_controller.rb

jessicafarias committed:
Railsy-er forms
Railsy-er form_with
Railsy-er form_for (commented)
Update new and create method in user_controller.rb

aracelicaes committed:
Creat Edit
Added route :edit
Updated controller with #edit
Adding html form for edit

jessicafarias committed:
Adding linters and fix offenses with rubocop

christianotieno commented on app/controllers/users_controller.rb:
```diff
@@ -0,0 +1,26 @@
+  def edit
+    @user = User.find(params[:id])
+  end
+
```
## Implement `update` action

* [x] finds user based on id from params, i.e. `@user = User.find(params[:id])` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_params.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))
* [x] `redirect_to edit_user_path` if user was saved
* [x] `render :edit` if user was not saved ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_update.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))

aracelicaes replied:
We have implemented this, however, #update doesn't render since we did not create a show.html.erb form, we will leave it as is since it's not a requirement.

aracelicaes replied:
_"Lastly, what if we could validate the email format by using either a regular expression or a built-in URI method?"_
Also, regarding the email validation, there is a RegEx existing and it is currently working. See **app/models/user.rb** **line 5**
```
validates :email,
            presence: true,
            uniqueness: true,
            format: { with: /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\z/i, on: :create }
```

christianotieno commented on config/routes.rb:
```diff
@@ -0,0 +1,3 @@
+Rails.application.routes.draw do
+  resources :users, only: [:new, :create, :edit]
```
## Add update to the routes 
* [x] `resources :users, only: [:new, :create, :edit, :update]` in config/routes.rb ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_route1.png) [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_route2.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))
```suggestion
  resources :users, only: [:new, :create, :edit, :update]
```

aracelicaes replied:
:update route has been added.

christianotieno reviewed (changes requested):
# REQUIRED CHANGES :warning:
Hey @jessicafarias & @aracelicaes  :wave:
-
You have done a splendid job! :tada: Keep it up.
We, however, do have some few changes to carry out before we proceed (see review comment).
##
* [x] Let's have a descriptive README.md ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/descriptive_readme.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))

##  \[OPTIONAL\] CHANGES
* [ ] Is possible to have a partial created for user form and reused in edit and new view? It would be nice to have. 😄 
* [ ] What of a form view to displays a list of the error messages (with #errors and full_messages usage) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_extra_shared_error.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))? 💭 
* [x] Lastly, what if we could alidate the email format by using either a regular expression or a built-in URI method? 😎 
##
- If you have any questions, let me know here or on slack and I will be glad to clarify with you. You can also reach me and connect with me on my social media or email. :eyes:
##
- When you have made the specified changes, please submit another code review. Thank you!
Happy coding :smile:
**[Christian Otieno](https://github.com/ChristianOtieno)**
[Twitter](https://twitter.com/iamchrisotieno) [linkedin](https://www.linkedin.com/in/christianotieno/) [GitLab](https://gitlab.com/christianotieno)
---
*P.S. all requested changes not marked by [OPTIONAL] or are not positive comments are **REQUIRED***

aracelicaes committed:
Add Update
Add :update to routes
Add #update to controller
Created new users & tried to edit.

aracelicaes committed:
Merge branch 'feature/re-former' of github.com:aracelicaes/re-former into feature/re-former

aracelicaes committed:
Made Corrections as Per TSE
Added update route
Added #update in Controller
Added ReadME

aracelicaes committed:
Rubocop fix
Fixed 1 rubocop error

maelfosso commented on app/controllers/users_controller.rb:
```diff
@@ -0,0 +1,38 @@
+    @user = User.find(params[:id])
+
+    if @user.update(user_params)
+      # flash[:success] = "User '#{@user}' succesfully updated!"
```
REmove comments, please.

maelfosso commented on app/controllers/users_controller.rb:
```diff
@@ -0,0 +1,38 @@
+      # flash[:success] = "User '#{@user}' succesfully updated!"
+      redirect_to edit_user_path
+    else
+      # flash[:error] = "Something went wrong"
```
Remove comments, please.

maelfosso reviewed (approved):
# Congratulations :clap: 

Thank you for applying the suggested changes.
You can merge it :+1: 
**PS**: Kindly remove the comments highlighted in comments below.

Good continuation :handshake: 

Mael FOSSO

jessicafarias committed:
Deleting comments
